### PR TITLE
Editing assignments as admin

### PR DESCRIPTION
**Pull request by Timmy-G**

### CODE REVIEW

File: server/controllers/admin.py [hunk 1]
```diff
@@ -124,6 +126,17 @@ def assignment(cid, aid):
     if assgn.course != current_course:
         return abort(401)
 
+    assgn.name = form.name._value()
```
Comment (Sumukh) [inline server/controllers/admin.py]: What happens if it's not a POST (or a valid submission?) - We want to validate that they submitted the form (and that the forms are valid). 

1. Method is POST. 
2. form.validate_on_submit()  (before you create the form from `assgn`) 

Take a look at `new_assignment`.

File: server/controllers/admin.py [hunk 2]
```diff
@@ -124,6 +126,17 @@ def assignment(cid, aid):
...
+    assgn.display_name = form.display_name._value()
+    assgn.due_date = datetime.strptime(form.due_date._value(), '%Y-%m-%d %H:%M:%S')       
+    assgn.lock_date = datetime.strptime(form.lock_date._value(), '%Y-%m-%d %H:%M:%S')
```
Comment (Sumukh) [inline server/controllers/admin.py]: I don't think you have to do this if you use form.validate_on_submit() and form.populate_obj 

Also note to self: These times should be converted to UTC sometime (but that's not related to this PR)

File: server/controllers/admin.py (hunk 1 above)
Comment (knrafto) [inline server/controllers/admin.py]: You only have to use `form.validate_on_submit()`. It'll check the request method is POST for you. 
Docs: https://flask-wtf.readthedocs.org/en/latest/quickstart.html#validating-forms

**Comment by Sumukh** [comment]
Great. I think we are going to add some cleanup regrading time conversion on forms to this PR (@Sumukh) 

We should also have tests for editing assignments.

File: server/controllers/admin.py [hunk 3]
```diff
@@ -117,13 +119,17 @@ def assignment(cid, aid):
     # Convert TZ to Pacific
     assgn.due_date = convert_to_pacific(assgn.due_date)
     assgn.lock_date = convert_to_pacific(assgn.lock_date)
```
Comment (knrafto) [inline server/controllers/admin.py]: WTForms should handle time conversion

Comment (Sumukh) [inline server/controllers/admin.py]: Any ideas here? I tried thinking of a few solutions but none of them seemed right. The best one I have so far is a custom DateTimeField renderer (in _formhelpers) that calls `utils.local_time`

I'm planning on leaving this for later.

**Comment by Sumukh** [comment]
Merged via #736

**Pull request closed without merging**